src/Connection/Internal/Http2ConnectionProcessor.php

```diff
@@ -103,15 +103,15 @@ final class Http2ConnectionProcessor implements Http2Processor
 
     private int|null $shutdown = null;
 
-    private Emitter $frameQueueSource;
+    private Queue $frameQueueSource;
 
     private Pipeline $frameQueue;
 
     public function __construct(EncryptableSocket $socket)
     {
         $this->socket = $socket;
         $this->hpack = new HPack;
-        $this->frameQueueSource = new Emitter();
+        $this->frameQueueSource = new Queue();
         $this->frameQueue = $this->frameQueueSource->pipe();
```

Comment:
It probably makes sense to rename `frameQueueSource` to `frameQueue` and `frameQueue` to `frameQueueIterator`